### PR TITLE
Find kokkos module

### PR DESCRIPTION
**Pull request by dalg24**
@masterleinad Would you review this please?

### CODE REVIEW

**Comment by masterleinad** [comment]
Sure.

**Comment by dalg24** [comment]
@aprokop I removed our patching Kokkos from the Dockerfile but I did not bother rebuilding.  I don't think it is necessary.

File: cmake/FindKokkos.cmake [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  set(Kokkos_DEFINITIONS ${PC_Kokkos_CFLAGS_OTHER})
+  set(Kokkos_DEVICES ${KOKKOS_GMAKE_DEVICES})
+  set(Kokkos_ARCH ${KOKKOS_GMAKE_ARCH})
+endif()
+
+# Clear the cache from all the mess in kokkos_generated_settings.cmake
+foreach(_var NVCC_WRAPPER KOKKOS_PATH KOKKOS_GMAKE_DEVICES KOKKOS_GMAKE_ARCH KOKKOS_DEBUG_CMAKE KOKKOS_GMAKE_USE_TPLS KOKKOS_CXX_STANDARD KOKKOS_GMAKE_OPTIONS KOKKOS_GMAKE_CUDA_OPTIONS KOKKOS_GMAKE_TPL_INCLUDE_DIRS KOKKOS_GMAKE_TPL_LIBRARY_DIRS KOKKOS_GMAKE_TPL_LIBRARY_NAMES KOKKOS_HEADERS KOKKOS_HEADERS_IMPL KOKKOS_HEADERS_CUDA KOKKOS_HEADERS_OPENMP KOKKOS_HEADERS_HPX KOKKOS_HEADERS_ROCM KOKKOS_HEADERS_THREADS KOKKOS_HEADERS_QTHREADS KOKKOS_OS KOKKOS_CPP_DEPENDS KOKKOS_LINK_DEPENDS KOKKOS_CXXFLAGS KOKKOS_CPPFLAGS KOKKOS_LDFLAGS KOKKOS_CXXLDFLAGS KOKKOS_LIBS KOKKOS_EXTRA_LIBS KOKKOS_LINK_FLAGS KOKKOS_INTERNAL_USE_CUDA KOKKOS_INTERNAL_USE_OPENMP KOKKOS_INTERNAL_USE_HPX KOKKOS_INTERNAL_USE_PTHREADS KOKKOS_INTERNAL_USE_SERIAL KOKKOS_INTERNAL_USE_ROCM KOKKOS_INTERNAL_USE_HPX KOKKOS_INTERNAL_USE_QTHREADS KOKKOS_SRC KOKKOS_CXX_FLAGS KOKKOS_CPP_FLAGS KOKKOS_LD_FLAGS KOKKOS_LIBS_LIST KOKKOS_EXTRA_LIBS_LIST KOKKOS_LINK_FLAGS)
```
Comment (aprokop) [inline cmake/FindKokkos.cmake]: Can you make this multiple lines for readability?

Comment (dalg24) [reply cmake/FindKokkos.cmake]: They are 45 variables, it will never be readable IMO

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,11 +1,12 @@
 cmake_minimum_required(VERSION 3.13)
 project(ArborX CXX)
 
-find_package(kokkos REQUIRED)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
```
Comment (masterleinad) [inline CMakeLists.txt]:
```suggestion
list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
```

File: cmake/FindKokkos.cmake [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  True if the system has the Kokkos library.
+``Kokkos_VERSION``
+  The version of the Kokkos library which was found.
+``Kokkos_INCLUDE_DIRS``
+  Include directories needed to use Kokkos.
+``Kokkos_LIBRARIES``
+  Libraries needed to link to Kokkos.
+``Kokkos_DEVICES``
+  Set of backends enabled.
+``kokkos_ARCH``
```
Comment (masterleinad) [inline cmake/FindKokkos.cmake]:
```suggestion
``Kokkos_ARCH``
```

**Comment by masterleinad** [comment]
I didn't test this, though.

File: cmake/FindKokkos.cmake (hunk 1 above)
Comment (aprokop) [reply cmake/FindKokkos.cmake]: OK. I assume there is no simple way to use regex (like `KOKKOS_*`)?

File: CMakeLists.txt (hunk 2 above)
Comment (dalg24) [reply CMakeLists.txt]: Any opinion on whether to prepend or append?

Comment (aprokop) [reply CMakeLists.txt]: Should not matter, but I would use prepend.

Comment (masterleinad) [reply CMakeLists.txt]: I mostly see projects appending, but maybe prepending is not a bad idea if we don't trust others writing a better `FindKokkos.cmake` file. I don't have a strong opinion here.

**Comment by dalg24** [comment]
Merge?

**Comment by aprokop** [comment]
@dalg24 With this merged PR, what's our minimum CMake version?

**Comment by dalg24** [comment]
> @dalg24 With this merged PR, what's our minimum CMake version?

I suspect 3.10 works

**Pull request merged**